RealmSwift/Geospatial.swift

```diff
@@ -0,0 +1,201 @@
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import CoreLocation
+import Realm
+import MapKit
+
+/**
+ A struct that represents the coordinates of a point formed by a latitude and a longitude value.
+
+  * Latitude ranges between -90 and 90 degrees, inclusive.
+  * Longitude ranges between -180 and 180 degrees, inclusive.
+ 
+ Values outside this ranges will return nil when trying to create a `GeoPoint`.
+
+ - note: There is no dedicated type to store Geospatial points, instead points should be stored as
+ [GeoJson-shaped](https://www.mongodb.com/docs/manual/reference/geojson/) embedded object, as explained below.
+
+ Persisting geo points in Realm is currently done using duck-typing, which means that any model class with a specific **shape**
+ can be queried as though it contained a geographical location.
+
+ the following is required:
+  * A String property with the value of **Point**: `@Persisted var type: String = "Point"`.
+  * A List containing a Longitude/Latitude pair: `@Persisted private var coordinates: List<Double>`.
+
+ The recommended approach is using an embedded object.
+ ```
+ public class Location: EmbeddedObject {
+   @Persisted private var coordinates: List<Double>
+   @Persisted private var type: String = "Point"
+
+   public var latitude: Double { return coordinates[1] }
+   public var longitude: Double { return coordinates[0] }
+
+   convenience init(_ latitude: Double, _ longitude: Double) {
+       self.init()
+       // Longitude comes first in the coordinates array of a GeoJson document
+       coordinates.append(objectsIn: [longitude, latitude])
+     }
+   }
+   ```
+
+ - warning: This structure cannot be persisted and can only be used to build other geospatial shapes
+   such as (`GeoBox`, `GeoPolygon` and `GeoCircle`).
+ */
+public typealias GeoPoint = RLMGeospatialPoint
+
+public extension GeoPoint {
+    /// Initialize a `GeoPoint`, with the specific values for latitude and longitude.
+    ///
+    /// Returns `nil` if the values of latitude and longitude are not within the ranges specified.
+    ///
+    /// - Parameter latitude: Latitude in degrees. Ranges between -90 and 90 degrees, inclusive.
+    /// - Parameter longitude: Longitude in degrees. Ranges between -180 and 180 degrees, inclusive.
+    convenience init?(_ latitude: Double, _ longitude: Double) {
```

Comment:
This is a place where argument labels are pretty important (since both arguments have the same type) and adding a convenience init without them is just making code using it less readable.